packages/flutter_tools/lib/src/commands/packages.dart

```diff
@@ -357,3 +349,60 @@ class PackagesGetCommand extends FlutterCommand {
     );
   }
 }
+
+// TODO: this doesn't seem right
+Pub get pub => context.get<Pub>()!;
+
+class PackagesPostGetHook extends FlutterCommand {
+  PackagesPostGetHook(this._context) {
+    argParser.addOption('directory', abbr: 'C', mandatory: true);
+    argParser.addFlag('example', defaultsTo: true);
+    argParser.addFlag('update-version-and-package-config');
+    argParser.addFlag('regenerate-platform-specific-tooling');
+  }
+  final PubContext _context;
+
+  @override
+  bool get hidden => true;
+
+  @override
+  // TODO: implement description
+  String get description => throw UnimplementedError();
+
+  @override
+  // TODO: implement name
+  String get name => '_post_pub_get';
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    // TODO: think about this part that actually happens before _runStdio (pre-get-hook?)
+    // - generateLocalizationsSyntheticPackage
+
+    final String directoryOption = argResults!['directory'] as String;
+    final String? target = findProjectRoot(globals.fs, directoryOption);
```

Comment:
Is this something that pub knows and could pass? I'm just thinking, if `target == null`, that probably means a bug on the tooling side, right?

In addition, I don't think it should be a toolExit, but maybe a `StateError`? ToolExit generally means that we do not believe there is a bug in the tool code (or its dependencies), but rather that the user is either providing invalid inputs (or the computer is in a bad state and needs to fixed).

Reply:
Oh wait, I read this wrong (before my coffee), we are basing this on a CLI option. So this means target will only be null if pub provided the wrong directory?

Reply:
Yup! So is the toolExit fine here?

Reply:
Oh, gotcha, that makes sense. 👌